File: Python/Find_Height_Of_Tree.py
class BinaryTree:
	def __init__(self, data):
		self.data = data
		self.left = None
		self.right = None
		
	def inOrder(self, tree):
		if tree is None:
			return
		self.inOrder(tree.left)
		print(tree.data, end=' ')
		self.inOrder(tree.right)
		
	def getHeight(self, root):
	       if root == None:
	       	return -1
	       else:
	       	return 1 + max( self.getHeight(root.left), self.getHeight(root.right) )
		
	def height(self):
		path = 0
		if root is None:
			return
		else:
			while self.right != None:
			     self = self.right
			     path = path +1
			     
			while self.left != None:
			     self = self.left
			     path1 = path1 +1
			     print(path1)
		
	def insertLeft(self, newNode):
		if self.left is None:
			self.left = BinaryTree(newNode)
		else:
			tree = BinaryTree(newNode)
			tree.left = self.left
			self.left = tree
		
	def insertRight(self, newNode):
		if self.right is None:
			self.right = BinaryTree(newNode)
		else:
			tree = BinaryTree(newNode)
			tree.right = self.right
			self.right = tree
		
root = BinaryTree(4)
root.insertLeft(8)
root.insertRight(5)
root.insertLeft(9)
root.insertRight(25)
root.insertLeft(29)
root.insertRight(45)

print('Inorder Traversal:')
root.inOrder(root)

print('\n')
treeHeight = root.getHeight(root)
print("Height of tree is:",treeHeight)

File: C Language/Find_Height_Of_Tree.c
#include<stdio.h>

struct node
{
    int data;
    struct node* left;
    struct node* right;
};

struct node* createNode(int item)
{
    struct node* newNode = (struct node*)malloc(sizeof(struct node));
    newNode -> data = item;
    newNode -> left = NULL;
    newNode -> right = NULL;
    return newNode;
}

void inOrder(struct node* root)
{
    if(root == NULL)
    {
        return;
    }
    inOrder(root -> left);
    printf("%d ", root -> data);
    inOrder(root -> right);
}

struct node* insertLeft(struct node* root, int item)
{
    root -> left = createNode(item);
    return root -> left;
}

struct node* insertRight(struct node* root, int item)
{
    root -> right = createNode(item);
    return root -> right;
}

int height(struct node* root)
{
    if(root == NULL)
    {
        return;
    }
    else
    {
        int left_side, right_side;
        left_side = height(root -> left);
        right_side = height(root -> right);
        if(left_side > right_side)
        {
            return left_side + 1;

        }
        else
        {
            return right_side + 1;
        }
    }
}

int main()
{
    struct node* root = createNode(2);
    insertLeft(root, 5);
    insertRight(root, 7);
    insertLeft(root -> left, 8);
    insertRight(root -> left, 4);
    insertLeft(root -> left -> left, 18);
    
    printf("Inorder Traversal:\n");
    inOrder(root);
    
    int treeHeight = height(root);
    printf("\n\nHeight of tree is: %d", treeHeight);
    return 0;
}